src/repo.rs

```diff
@@ -2346,6 +2347,24 @@ impl Repository {
         }
         Ok(ignored == 1)
     }
+
+    /// Perform a cherrypick
+    pub fn cherrypick(
+        &self,
+        commit: &Commit<'_>,
+        options: Option<&mut CherrypickOptions<'_>>,
+    ) -> Result<(), Error> {
+        let raw_opts = options.map(|o| o.raw());
+        let ptr_raw_opts = match raw_opts {
```

Comment:
Oh this'll want to be `raw_opts.as_ref()` to avoid moving `raw_opts` and creating a dangling pointer again

Reply:
Yes, this as_ref() method was what I was missing. Done